components/search/index.tsx

```diff
@@ -1,49 +1,52 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {getMorePostsForSearch} from 'mattermost-redux/actions/search';
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {AnyAction, bindActionCreators, Dispatch} from 'redux';
 
 import {
-    updateSearchTerms,
-    showSearchResults,
-    showMentions,
-    showFlaggedPosts,
-    closeRightHandSide,
-    updateRhsState,
+    updateSearchTerms, showSearchResults, showMentions, showFlaggedPosts, closeRightHandSide, updateRhsState, setRhsExpanded,
 } from 'actions/views/rhs';
 import {autocompleteChannelsForSearch} from 'actions/channel_actions';
 import {autocompleteUsersInTeam} from 'actions/user_actions';
-import {getRhsState, getSearchTerms, getIsSearchingTerm, getIsRhsOpen} from 'selectors/rhs';
+import {getRhsState, getSearchTerms, getIsSearchingTerm, getIsRhsOpen, getIsRhsExpanded} from 'selectors/rhs';
 import {RHSStates} from 'utils/constants';
+import {GlobalState} from 'types/store';
 
-import SearchBar from './search_bar.jsx';
+import Search from './search';
+import type {StateProps, DispatchProps, OwnProps} from './types';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: GlobalState) {
     const rhsState = getRhsState(state);
 
     return {
+        isRhsExpanded: getIsRhsExpanded(state),
+        isRhsOpen: getIsRhsOpen(state),
         isSearchingTerm: getIsSearchingTerm(state),
         searchTerms: getSearchTerms(state),
+        searchVisible: Boolean(rhsState) && rhsState !== RHSStates.PLUGIN,
         isMentionSearch: rhsState === RHSStates.MENTION,
         isFlaggedPosts: rhsState === RHSStates.FLAG,
-        isRhsOpen: getIsRhsOpen(state),
+        isPinnedPosts: rhsState === RHSStates.PIN,
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
     return {
         actions: bindActionCreators({
             updateSearchTerms,
             showSearchResults,
             showMentions,
             showFlaggedPosts,
+            setRhsExpanded,
             closeRightHandSide,
             autocompleteChannelsForSearch,
             autocompleteUsersInTeam,
             updateRhsState,
+            getMorePostsForSearch,
         }, dispatch),
     };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
+export default connect<StateProps, DispatchProps, OwnProps, GlobalState>(mapStateToProps, mapDispatchToProps)(Search);
```

Comment:
It is not ultimately necessary for typescript, but it leaves a vulnerability if not.

I was having your (https://developers.mattermost.com/contribute/webapp/migrating-to-typescript/)[migrating to typescript] guide in mind

Reply:
Fair enough. I don't think we need to require them since it adds more overhead to the TypeScript migration on top of what people already need to understand about it, but it certainly doesn't hurt to have here.